### PR TITLE
Add --verbose option to 'stg new'

### PR DESCRIPTION
**Pull request by topher200**
Git has a great feature where you can get a 'git diff' in the editor window when
creating a commit. This can help give context while you write your commit
message.

This PR adds this functionality to 'stg new'. Just like Git, there's two ways to
trigger the new feature:
1. `stg new --verbose`
2. Setting the config setting `commit.verbose=1` (borrows Git's config var)

[asciicast](https://asciinema.org/a/422507)

### CODE REVIEW

**Comment by topher200** [comment]
For reference, here's what standard Git uses for commit messages of various verbosities:

## Default (no `--verbose`):

```
<commit message here>
# Please enter the commit message for your changes. Lines starting
# with '#' will be ignored, and an empty message aborts the commit.
#
# On branch stgit-topher
# Changes to be committed:
#       modified:   README.md
#       new file:   asdf
#
# Untracked files:
#       q
#
```

## `--verbose`: 
Help message:
> Show unified diff between the HEAD commit and what would be committed at the bottom of the commit message template to help the user describe the commit by reminding what changes the commit has. Note that this diff output doesn’t have its lines prefixed with #. This diff will not be a part of the commit message. See the commit.verbose configuration variable in git-config(1).

**tldr: show the output of `git diff --cached`**

```
<commit message here>
# Please enter the commit message for your changes. Lines starting
# with '#' will be ignored, and an empty message aborts the commit.
#
# On branch stgit-topher
# Changes to be committed:
#       new file:   asdf
#
# Changes not staged for commit:
#       modified:   README.md
#
# Untracked files:
#       q
#
# ------------------------ >8 ------------------------
# Do not modify or remove the line above.
# Everything below it will be ignored.
diff --git a/asdf b/asdf
new file mode 100644
index 0000000..e69de29
```

## `--verbose --verbose`:

> If specified twice, show in addition the unified diff between what would be committed and the worktree files, i.e. the unstaged changes to tracked files.

**tldr: show the output of `git diff`**
```
<commit message here>
# Please enter the commit message for your changes. Lines starting
# with '#' will be ignored, and an empty message aborts the commit.
#
# On branch stgit-topher
# Changes to be committed:
#       new file:   asdf
#
# Changes not staged for commit:
#       modified:   README.md
#
# Untracked files:
#       q
#
# ------------------------ >8 ------------------------
# Do not modify or remove the line above.
# Everything below it will be ignored.
#
# Changes to be committed:
diff --git c/asdf i/asdf
new file mode 100644
index 0000000..e69de29
# --------------------------------------------------
# Changes not staged for commit:
diff --git i/README.md w/README.md
index 3f9686f..8bd6648 100644
--- i/README.md
+++ w/README.md
```

## Summary
Git has three options for `verbose`: 0, 1, 2. I don't think that this distinction makes sense for stgit! Git's commit message comes into play when one is about to make a commit, which means that the 'staged' files matter a ton. That's not the case for `stg new`! stgit isn't about to make a commit from the staged files -- that will happen later with `stg refresh`. Therefore, I think that a binary `--verbose` option is best for stgit and that we should just show the output of `git diff` always.

**Comment by jpgrayson** [comment]
I buy this.
